anomalib/models/components/base/anomaly_module.py

```diff
@@ -82,6 +84,8 @@ def predict_step(self, batch: Any, batch_idx: int, _dataloader_idx: Optional[int
         outputs["pred_labels"] = outputs["pred_scores"] >= self.image_threshold.value
         if "anomaly_maps" in outputs.keys():
             outputs["pred_masks"] = outputs["anomaly_maps"] >= self.pixel_threshold.value
+            if "pred_boxes" not in outputs.keys():
+                outputs["pred_boxes"] = masks_to_boxes(outputs["pred_masks"])
```

Comment:
Is it possible to add `pred_boxes` to `outputs` only when the task type is `detection`? Or am I missing something?

Reply:
The problem is that we can't access task type information from within the lightning module. This is one of the reasons why we should consider switching to custom training loops.